src/google/protobuf/compiler/php/php_generator.cc

```diff
@@ -267,6 +271,18 @@ std::string GeneratedEnumFileName(const EnumDescriptor* en,
   return result + ".php";
 }
 
+std::string GeneratedServiceFileName(const ServiceDescriptor* service,
```

Comment:
I just pushed a commit to use `FullClassName` to generate the service file name.  I had to overload the method because `ServiceDescriptor` does not have a `containing_type` method.  I also extracted a `NamespacedName` function to allow re-using the logic.